mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -63,62 +69,71 @@ public class ExecutorStateManagerImpl implements ExecutorStateManager {
         .build();
 
     @Override
-    public void putIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state) {
+    public void trackIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state) {
         this.taskExecutorStateMap.putIfAbsent(taskExecutorID, state);
         if (this.archivedState.getIfPresent(taskExecutorID) != null) {
             log.info("Reviving archived executor: {}", taskExecutorID);
             this.archivedState.invalidate(taskExecutorID);
         }
 
-        // add to buckets. however new executors won't have valid registration at this moment and requires marking
-        // again later when registration is ready.
+        tryMarkAvailable(taskExecutorID, state);
```

Comment:
nit:
I would change this code as below.
```
if (state.isAvailable()) {
  Preconditions.checkState(markAsAvailable());
}
```

Reply:
we don't need to fail if the given state is not available yet right? (checkState throws exception)